### ISSUE TITLE
Processing Agilent DTIM-MS data

### ISSUE THREAD

**Issue by MaxFeuerstein** [mzmine/mzmine3#766]
**Is your feature request related to a problem? Please describe.**
I am mainly working with IM-MS data from Agilent 6560  DTIM-QTOF and tested your software for IM-AI data.
Peak picking on MS1 level seems to work well, but I have troubles extracting MS2-type specta.
It's possible that this issue is related to an issue with the workflow I used, but I wanted to ask if IM-AI data can be processed and if not if you could consider implementation in a newer version?
I am happy to send some example files if this helps.

**Describe the solution you'd like**
If possible, direct import of Agilent .d data files would be great!

**Describe alternatives you've considered**
Coversion to mzML is not a big issue, but the possibility for direct import is always a plus.

**Additional context**
IM-AI data in DTIM-MS is similar to HDMSE data for the TWIM system.
However, when implementing MS2 extraction is important to keep in mind that there is a small offset in arrival times of fragment ions compared to precursor ions. They experience an acceleration during fragmentation and the apparent arrival times are ~0.1-0.2 ms earlier than precursor ions. 
An option to set this offset would be nice and is for example implemented in Skyline.

Thanks a lot for your great work, I really appreciate the modularity and the visualization provided by mzMine3!!
I am already looking forward to the "final" documentation, as this might already answer some of my questions.
Thanks a lot in advance and best regards,

Max

**Comment by SteffenHeu** [comment]
Hi Max,

MZmine is mainly developed as a software for DDA acquisitions. However, i made some tests for a simple DIA correlation based on MS1 peak shape and mobility windows of precursors and fragments. Since we mainly use PASEF in our lab, i lack the data for proper testing. If you are interested, you can play around with the experimental version here: https://github.com/mzmine/mzmine3/actions/runs/2629111207. Feedback would be welcome. It requires mobility scan merging for MS1 and MS2 frames. The MS2 can be created by the module shown here: (only available in the version in the link above)
![grafik](https://user-images.githubusercontent.com/37407705/177761285-c802d96b-ce70-4672-8f16-c1a4ea4a01ce.png)

**Comment by MaxFeuerstein** [comment]
Hi Steffen, 

the DIA assignment for MS2 is working reasonably well with the Agilent data. 
MS2 deconvolution in MS-DIAL seems to still work slightly better, but  I am quite used to work with MS-DIAL and the results for mzMine will probably improve after further optimizing the method parameters.

I really appreciate your correlation-based algorithm for MS2 extraction/assignment in combination with the visualization options and I think the DIA feature will be quite useful for DIA-type IM-MS workflows.,
It would be great if it could be implemented as a "standard" option in future mzMine releases!

I am looking forward to updates of software and documentation and will send more feedback after additional tests.

Thanks a lot and best regards,

Max

**Comment by SteffenHeu** [comment]
Hey Max,
thanks a lot for the feedback, i really appreciate it. Do you think there is something to be fixed or could this be added as a beta feature in the next MZmine release?

Do you think an additional visualisation would be useful? If yes, which one?

**Comment by MaxFeuerstein** [comment]
Hi Steffen, 

I think this would be great, as more feedback from other users might also help to improve the DIA workflow.
To quickly evaluate the MS2-deconvolution it would be useful to show "raw" MS2 (i.e., only filtered by predefined RT and arrival time) vs. "clean" spectra using Pearson correlation etc.
Some additional documentation might be necessary to fully evaluate the workflow, as it is not 100% clear how to use a few of the settings. For example, the "number of correlated points" is a little bit unclear.
Is the Pearson correlation based on LC-information (+mobility dependent filters) or based on mobility data (within a certain RT range)?

What is the recommended order of steps for the workflow?
I tried to build a DIA workflow based on your comments above and the "default" IM-MS workflow in the processing wizzard (see attached screenshot), but I'm not 100% sure if the order of steps I used makes sense

Another issue that was already reported by another user is that mzMine creates temp files (for IM-data ~1 GB per file) in a ...AppData/Local/temp which are not deleted after closing the program. 
For only running small batches it is fine to delete the files after closing the program, but for larger batches this might be an issue.

Thanks a lot and best regards,

Max

![mzMIne_DIA_workflow](https://user-images.githubusercontent.com/11314509/178950216-16a5acc5-5d75-4714-8f68-b3c724db78e9.PNG)

**Comment by SteffenHeu** [comment]
> To quickly evaluate the MS2-deconvolution it would be useful to show "raw" MS2 (i.e., only filtered by predefined RT and arrival time) vs. "clean" spectra using Pearson correlation etc.

Would a mirror window be sufficient? So one spectrum should be filtered by mobility of the precursor, but include all ions that elute at the same retention time and another should be the "clean" MS/MS?

> Some additional documentation might be necessary to fully evaluate the workflow, as it is not 100% clear how to use a few of the settings. For example, the "number of correlated points" is a little bit unclear.
Is the Pearson correlation based on LC-information (+mobility dependent filters) or based on mobility data (within a certain RT range)?

I agree, since it was just my testing version i did not create a documentation yet. To answer your question: The pearson correlation is based on the Peak shape in retention time dimension. The module will create MS2 ion chromatograms and correlate their shape (intensity vs rt) with the one of the precursor. To be an eligible fragment, the MS2-ion must also be detected in the drift spectra within the mobility/drift time FWHM of the precursor ion. I guess this should allow for a big enough shift due to the ion acceleration, or do you have other experiences?


> Another issue that was already reported by another user is that mzMine creates temp files (for IM-data ~1 GB per file) in a ...AppData/Local/temp which are not deleted after closing the program.
For only running small batches it is fine to delete the files after closing the program, but for larger batches this might be an issue.

Unfortunately, this is an issue we are aware of, but cannot fix at the moment. Windows does not "release" the files until the process that created them is closed. We can force the deletion, but then we will run into issues with multiple instances of MZmine running in parallel, since we delete their temp files aswell. This is not an issue in Linux or Mac.

**Comment by MaxFeuerstein** [comment]
Yes, I think such a mirror window would be great help to quickly get an impression of how useful the set parameters are.

OK, thanks for the explanation! The small difference is probably included in an appropriate tolerance window for the mobility dimension. The shift of peak apex of precursor and fragment ions is typically way less than the FWHM and the fragment peaks still are within the range of the precursor peak. 

Thanks a lot for the info!

![image](https://user-images.githubusercontent.com/11314509/179562350-6a8e63fa-a7aa-47cf-aa87-614584113c19.png)

**Comment by SteffenHeu** [comment]
@MaxFeuerstein This has been merged to the master branch now. Feel free to provide any further feedback. The visualizer is accessible from the feature table

**Comment by alnocete** [comment]
Hi Steffen. Thank you for this.
I will test this and will give you a feedback afterwards.

Best,
Ann
